gitpod-io/gitpod
# Automatically cancel org's subscription upon its deletion

## Description

When an org is deleted (like if you remove it manually or your user gets deleted), all of its corresponding subscriptions in Stripe should be cancelled to reflect the deletion. This PR takes care of that.

## Related Issue(s)

Fixes ENT-215

## How to test

https://ft-cancel-89c45b748f.preview.gitpod-dev.com/workspaces

1. In the preview environment, create a new org (not the one that gets created automatically, so that you can better observe the change after it gets deleted), go to billing and upgrade your org to a PAYG plan (use details as described [here](https://docs.stripe.com/testing#testing-interactively))
2. Log into our Stripe testing instance to verify the subscription (under https://dashboard.stripe.com/test/subscriptions)
3. Delete your org
4. Verify in Stripe that the subscription is no longer marked as `Active`.

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
